internal/rust/scanner/src/lib.rs

```diff
@@ -15,12 +15,20 @@ pub struct Scanner {
     token: T,
     ts: u32,
     te: u32,
+    lines: Vec<u32>,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct Position {
+    pub line: u32,
+    pub column: u32,
 }
 
 #[derive(Debug, PartialEq, Clone)]
 pub struct Token {
     pub tok: T,
     pub lit: String,
+    pub pos: u32,
```

Comment:
It seems like instead of storing the file offset inside the token, you could just keep the start `Postition` (including the line and column) and the end `Position` directly in the token returned from `scan()`.

Then you wouldn't need to do the mapping from file offset to `Position` which does a binary search for each call to `pos()` and has (log n) performance.

We discussed how this might increase the memory footprint since tokens may be cloned frequently.  It seems like if tokens are immutable, this could be avoided.

Can you file an issue for this?

Reply:
done https://github.com/influxdata/flux/issues/1608.

Why do you think that if tokens are immutable this could be avoided?

Maybe we should just change the implementation to work with references (ownership problems) in order to avoid cloning, given that tokens are immutable.

Reply:
Yeah, I just mean that if an instance of token never changes, then only one instance needs to be created (no cloning needed), since all the readers can share the same instance.  Then you could just use references everywhere.